w3af/core/ui/api/resources/scans.py

```diff
@@ -102,21 +104,28 @@ def start_scan():
     except BaseFrameworkException, bfe:
         abort(400, str(bfe))
 
+    scan_id = get_new_scan_id()
+    scan_info = ScanInfo()
+    scan_info.w3af_core = w3af_core
+    scan_info.target_urls = target_urls
+    scan_info.temp_profile_path = scan_profile_file_name
+    scan_info.output = RESTAPIOutput()
+    SCANS[scan_id] = scan_info
+
+    # Clear all current output plugins
+    om.manager.set_output_plugins([])
+    # Add the REST API output plugin
+    om.manager.set_output_plugin_inst(scan_info.output)
+
     #
     # Finally, start the scan in a different thread
     #
-    scan_id = get_new_scan_id()
-    scan_info_setup = Event()
-
-    args = (target_urls, scan_profile, scan_info_setup)
```

Comment:
Why did you remove this Event? I don't remember exactly why I added it, but it seems that at some point it made sense to me to use it to:

```python
# Wait until the thread starts
scan_info_setup.wait()
```